cumulus/client/relay-chain-rpc-interface/src/reconnecting_ws_client.rs

```diff
@@ -460,4 +496,48 @@ mod test {
 			url_to_string_with_port(url)
 		);
 	}
+
+	#[tokio::test]
+	// Testing the retry logic at full means increasing CI with half a minute according
+	// to the current logic, so lets test it best effort.
+	async fn client_manager_retry_logic() {
+		let port = portpicker::pick_unused_port().unwrap();
+		let server = jsonrpsee::server::Server::builder()
+			.build(format!("0.0.0.0:{}", port))
+			.await
+			.unwrap();
+
+		// Wait three seconds while attempting connection.
+		let conn_res = tokio::spawn(async move {
+			tokio::time::timeout(
+				Duration::from_secs(3),
+				ClientManager::new(vec![format!("ws://127.0.0.1:{}", port)]),
+			)
+			.await
+		});
+
+		// Start the server too.
+		let server = tokio::spawn(async {
+			tokio::time::sleep(Duration::from_secs(10)).await;
+			server.start(Methods::default())
+		});
+
+		// By this time the client can not make a connection because the server is not up.
+		assert!(conn_res.await.unwrap().is_err());
+
+		// Trying to connect again to the RPC with a client that stays around for sufficient
+		// time to catche the RPC server online and connect to it.
+		let conn_res = tokio::spawn(async move {
+			tokio::time::timeout(
+				Duration::from_secs(8),
```

Comment:
CI tests are run on quit overloaded machines. 1s margin may not be enough, but let's see how it goes.

Reply:
Pushed a simpler version of the test here: [ee68d04](https://github.com/paritytech/polkadot-sdk/pull/5515/commits/ee68d040653b801897af8e646c19508a8015511d).